src/test/calculate.test.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import calculate from 'logic/calculate';
+import Span from 'components/Span';
+
+test('render', () => {
+  const obj = {
+    total: '20',
+    next: '2',
+    operation: '-',
+  };
+  render(<Span obj={obj} setObj={() => obj} text="=" color="white" />);
+  const equal = calculate(obj, '=');
+  const button = screen.getByRole('button');
+  fireEvent.click(button);
+  expect(equal.total).toBe('18');
+});
+test('render', () => {
+  const obj = {
+    total: '2',
+    next: '2',
+    operation: '+',
+  };
+  render(<Span obj={obj} setObj={() => obj} text="=" color="white" />);
+  const equal = calculate(obj, '=');
+  const button = screen.getByRole('button');
+  fireEvent.click(button);
+  expect(equal.total).toBe('4');
+});
```

Comment:
- [ ] Again here the requirement is that we have to test for all scenarios of the `calculate.js` component, These scenarios include:

- What happens when a user clicks the `AC` button to reset the calculator
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/eb58b055-a065-4f38-a868-e7f24e9dcd1c)

- What happens if there is an operation
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/27ef786c-e033-49f0-a4cb-d72960c03aee)

- What happens if there is no operation
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/641e9062-3ca5-4b33-9b21-7c83674f5c9f)

- What happens when a user clicks the  `.` point operand
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/46ea9ffc-a8a1-436b-afd9-ff351858fbc9)

- What happes when a user clicks the `-/+` minus/plus sign
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/f657e050-ec91-4af8-ad3d-c41d4cf29a87)

- What happens when a user pressed an operation after pressing the `=` sign
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/0b0a0a9f-1f48-4695-a8c6-286326646ae8)

- What happens when a user pressed an operation button and there is an existing operation
![image](https://github.com/tumainimaganiko/math-magicians/assets/81932478/7ae005a8-b3c6-4234-a608-688a9a2d458f)

-